src/modules/expeditions/services/dailyFragments/DailyFragments.types.ts

```diff
@@ -1,14 +1,20 @@
+import { IVisit } from '../../interfaces/IVisit.interface';
+import { AddressWithId } from '../../interfaces/shared';
 import { VisitModel } from '../../models/Visit.model';
 
 export enum DailyFragmentsTypes {
   VISIT = 'DAILY_VISIT',
 }
 
+export interface DailyRewardsBaseParams extends AddressWithId {
+  type: DailyFragmentsTypes;
+}
+
 export interface DailyFragmentsServiceParams {
   visitModel: VisitModel;
 }
 
-export interface DailyFragments {
-  allVisits: number;
-  lastVisit: number | Date;
-}
+export type DailyFragments = Pick<IVisit, 'allVisits' | 'fragments'> & {
+  lastVisit: Date | number;
+  nextVisit: Date;
```

Comment:
I find it more handy to handle this on backend rather than calculating it again with the same function on the front-end. I'm ok with removing it if you disagree.